### PR TITLE
Add repeat option to `SpriteBase3D`

### LINKED ISSUE #82044: Sprite3D texture wrapping causing pixel lines on border
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2080 Ti (NVIDIA; 31.0.15.3734) - Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 Threads)

### Issue description

Sprite3D has pixel lines which appear to be the texture wrapping. There are no options on Sprite3D to disable wrapping.

image

Some troubleshooting for this issue has been carried out in my [Reddit thread.](https://old.reddit.com/r/godot/comments/16nobpu/new_to_godot_and_loving_it_any_idea_how_i_can/)

Tested on 4.1.1 and 4.2dev5

### Steps to reproduce

1. Create Sprite3D node
2. Import texture with lossless compression (or any compression, nothing I tried made a difference)
3. Apply texture to Sprite3D

### Minimal reproduction project

[Sprite3D_bug_21-09-23.zip](https://github.com/godotengine/godot/files/12684098/Sprite3D_bug_21-09-23.zip)

### PR DESCRIPTION
**Pull request by imaskillissue**
Fixes #82044

Added to SpiteBase3D to Flags Repeat.

### CODE REVIEW

File: scene/3d/label_3d.cpp [hunk 1]
```diff
@@ -379,7 +379,7 @@ void Label3D::_generate_glyph_surfaces(const Glyph &p_glyph, Vector2 &r_offset,
 			}
 
 			RID shader_rid;
-			StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, msdf, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), get_draw_flag(FLAG_FIXED_SIZE), texture_filter, alpha_antialiasing_mode, &shader_rid);
+			StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, msdf, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), true,get_draw_flag(FLAG_FIXED_SIZE), texture_filter, alpha_antialiasing_mode, &shader_rid);
```
Comment (AThousandShips) [inline scene/3d/label_3d.cpp:382]:
```suggestion
			StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, msdf, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), true, get_draw_flag(FLAG_FIXED_SIZE), texture_filter, alpha_antialiasing_mode, &shader_rid);
```

File: scene/3d/sprite_3d.cpp [hunk 2]
```diff
@@ -264,7 +264,7 @@ void SpriteBase3D::draw_texture_rect(Ref<Texture2D> p_texture, Rect2 p_dst_rect,
 	}
 
 	RID shader_rid;
-	StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, false, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), get_draw_flag(FLAG_FIXED_SIZE), get_texture_filter(), alpha_antialiasing_mode, &shader_rid);
+	StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, false, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), get_draw_flag(FLAG_REPEAT),get_draw_flag(FLAG_FIXED_SIZE), get_texture_filter(), alpha_antialiasing_mode, &shader_rid);
```
Comment (AThousandShips) [inline scene/3d/sprite_3d.cpp:267]:
```suggestion
	StandardMaterial3D::get_material_for_2d(get_draw_flag(FLAG_SHADED), mat_transparency, get_draw_flag(FLAG_DOUBLE_SIDED), get_billboard_mode() == StandardMaterial3D::BILLBOARD_ENABLED, get_billboard_mode() == StandardMaterial3D::BILLBOARD_FIXED_Y, false, get_draw_flag(FLAG_DISABLE_DEPTH_TEST), get_draw_flag(FLAG_REPEAT), get_draw_flag(FLAG_FIXED_SIZE), get_texture_filter(), alpha_antialiasing_mode, &shader_rid);
```

**Comment by imaskillissue** [comment]
Should work now

**Comment by Calinou** [comment]
Note that since this uses a boolean property, it means it won't be able to address https://github.com/godotengine/godot/issues/51625 unless we switch this property to an enum later on.

**Comment by imaskillissue** [comment]
> Note that since this PR uses a boolean flag property, this means it won't be able to address https://github.com/godotengine/godot/issues/51625 unless we switch this property to an enum later on. This is probably not an issue (existing boolean values would map to the enum values "Disabled" and "Enabled" respectively), but I'm pointing it out just in case.

If you want I could change it later I did it like all the other flags

**Comment by clayjohn** [comment]
We discussed this in today's rendering meeting (sorry for the long wait). We realized that we don't actually need to expose the repeat setting to users, as we already know when users are relying on the repeat property. To repeat a texture on a Sprite 3D, you have to enable `region` and set it to a region larger than the size of the texture. 

Using that knowledge, we can automatically enable the repeat filter for the user instead of having them enable/disable it manually.

**Comment by clayjohn** [comment]
Superseded by https://github.com/godotengine/godot/pull/90202

Thank you for your contribution anyway! Sorry that we weren't able to get this into a mergeable state

**Pull request closed without merging**